connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -296,7 +296,7 @@ public boolean contains(String connector) {
      */
     @Override
     public void putConnectorConfig(String connector, Map<String, String> properties) {
```

Comment:
Yeah, it would be possible to do that, and that object would have to:
1. Look in the configuration for the `connector.class` property
2. Use the plugins mechanism to load the specified connector class
3. Get the ConfigDef from the connector instance
4. Create a `ConnectorConfig` instance using the ConfigDef
5. Use the `ConnectorConfig.toString()` method to mask any PASSWORD fields.

The problem is that some connectors may include secrets in configs that are not of type PASSWORD, and this approach would still not mask those secrets. For example, JDBC URLs can contain the username and password, but it's not likely that any connector that has a JDBC URL as a config treats this as a PASSWORD field.

Reply:
Yeah, I was thinking something simpler. Like having an `UnvalidatedConnectorConfig` which just wrapped the map we have here. We could override `toString` and document the potential sensitivity of the configs. Anyway, up to you if you think it would be helpful or not.

Reply:
I'm not sure it adds much value. I'm not sure we can print any of the connector configuration properties if we don't know which connector-specific properties are sensitive.

Reply:
Fair enough.